stack/dashboard/rpm/wazuh-dashboard.spec

```diff
@@ -138,6 +138,14 @@ if [ ! -f %{INSTALLATION_DIR}/config/opensearch_dashboards.keystore ]; then
   runuser %{USER} --shell="/bin/bash" --command="echo kibanaserver | %{INSTALL_DIR}/bin/opensearch-dashboards-keystore add opensearch.username --stdin" > /dev/null 2>&1
   runuser %{USER} --shell="/bin/bash" --command="echo kibanaserver | %{INSTALL_DIR}/bin/opensearch-dashboards-keystore add opensearch.password --stdin" > /dev/null 2>&1
 fi
+
+if [ -f /etc/os-release ]; then
+  source /etc/os-release
+  if [ "${NAME}" = "Red Hat Enterprise Linux" ] && [ "$((${VERSION_ID:0:1}))" -ge 9 ]; then
+    rm -f /etc/init.d/wazuh-dashboard
```

Comment:
Use `%{_initrddir}`

Reply:
Done in https://github.com/wazuh/wazuh-packages/pull/1898/commits/ee01caf18e2836e56db4907b1b7be17bb97f5ed1

<details><summary>Wazuh dashboard build</summary>

```
╰─➤  bash build_package.sh -r wp.1880                              
Sending build context to Docker daemon  14.34kB
Step 1/8 : FROM rockylinux:8.5
 ---> 210996f98b85
Step 2/8 : RUN yum clean all && yum update -y
 ---> Using cache
 ---> e5b2b0c56801
Step 3/8 : RUN yum install -y     curl     tar     findutils     git     xz      gcc     make     bc     sed     gzip     autoconf     automake     libtool     jq
 ---> Using cache
 ---> 82b6d14bf4b7
Step 4/8 : RUN git clone https://github.com/google/brotli.git
 ---> Using cache
 ---> 1331c1f010b7
Step 5/8 : RUN cd brotli && ./bootstrap && ./configure --prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --libexecdir=/usr/lib64/brotli --libdir=/usr/lib64/brotli --datarootdir=/usr/share --mandir=/usr/share/man/man1 --docdir=/usr/share/doc     && make && make install
 ---> Using cache
 ---> d41c1160474c
Step 6/8 : ADD builder.sh /usr/local/bin/builder
 ---> Using cache
 ---> d149e25bfb0b
Step 7/8 : RUN chmod +x /usr/local/bin/builder
 ---> Using cache
 ---> 3a4b20575b0b
Step 8/8 : ENTRYPOINT ["/usr/local/bin/builder"]
 ---> Using cache
 ---> 5320fa697e2d
Successfully built 5320fa697e2d
Successfully tagged dashboard_base_builder:latest
Removing queryWorkbenchDashboards...
Plugin removal complete
Removing anomalyDetectionDashboards...
Plugin removal complete
Removing observabilityDashboards...
Plugin removal complete
Base file wazuh-dashboard-base-4.4.0-wp.1880-linux-x64.tar.xz added to /stack/dashboard/rpm/output.
Sending build context to Docker daemon  4.608kB
Step 1/6 : FROM centos:7
 ---> eeb6ee3f44bd
Step 2/6 : RUN yum clean all && yum update -y
 ---> Using cache
 ---> 9e1308bcf08f
Step 3/6 : RUN yum install -y openssh-clients sudo gnupg     yum-utils epel-release redhat-rpm-config rpm-devel     zlib zlib-devel rpm-build autoconf automake     glibc-devel libtool perl
 ---> Using cache
 ---> c88764fc8a5f
Step 4/6 : ADD builder.sh /usr/local/bin/builder
 ---> Using cache
 ---> 07ee1a63d3cc
Step 5/6 : RUN chmod +x /usr/local/bin/builder
 ---> Using cache
 ---> 981089e6483f
Step 6/6 : ENTRYPOINT ["/usr/local/bin/builder"]
 ---> Using cache
 ---> fb5d5816e32a
Successfully built fb5d5816e32a
Successfully tagged rpm_dashboard_builder_x86:latest
Building target platforms: x86_64
Building for target x86_64
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.ivxgC8
+ umask 022
+ cd /build/rpmbuild/BUILD
+ cp /tmp/wazuh-dashboard-base-4.4.0-wp.1880-linux-x64.tar.xz ./
+ groupadd wazuh-dashboard
+ useradd -g wazuh-dashboard wazuh-dashboard
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.pb8RSv
+ umask 022
+ cd /build/rpmbuild/BUILD
+ tar -xf wazuh-dashboard-base-4.4.0-wp.1880-linux-x64.tar.xz
+ exit 0
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.u5VxFf
+ umask 022
+ cd /build/rpmbuild/BUILD
+ '[' /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64 '!=' / ']'
+ rm -rf /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64
++ dirname /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64
+ mkdir -p /build/rpmbuild/BUILDROOT
+ mkdir /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64
+ mkdir -p /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/wazuh-dashboard
+ mkdir -p /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard
+ mkdir -p /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/systemd/system
+ mkdir -p /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/rc.d/init.d
+ mkdir -p /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/default
+ cp wazuh-dashboard-base/etc/node.options /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/wazuh-dashboard
+ cp wazuh-dashboard-base/etc/opensearch_dashboards.yml /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/wazuh-dashboard
+ cp wazuh-dashboard-base/VERSION /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard
+ mv wazuh-dashboard-base/LICENSE.txt wazuh-dashboard-base/NOTICE.txt wazuh-dashboard-base/README.txt wazuh-dashboard-base/VERSION wazuh-dashboard-base/assets wazuh-dashboard-base/bin wazuh-dashboard-base/config wazuh-dashboard-base/data wazuh-dashboard-base/etc wazuh-dashboard-base/manifest.yml wazuh-dashboard-base/node wazuh-dashboard-base/node_modules wazuh-dashboard-base/package.json wazuh-dashboard-base/plugins wazuh-dashboard-base/src /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard
+ mkdir -p /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/config
+ cp /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/etc/services/wazuh-dashboard.service /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/systemd/system/wazuh-dashboard.service
+ cp /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/etc/services/default /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/default/wazuh-dashboard
+ cp /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/etc/services/wazuh-dashboard /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/rc.d/init.d/wazuh-dashboard
+ chmod 750 /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/rc.d/init.d/wazuh-dashboard
+ chmod 640 /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/systemd/system/wazuh-dashboard.service
+ chmod 640 /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/default/wazuh-dashboard
+ rm -rf /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/etc/
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard -exec chown wazuh-dashboard:wazuh-dashboard '{}' ';'
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/wazuh-dashboard -exec chown wazuh-dashboard:wazuh-dashboard '{}' ';'
+ chown root:root /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/systemd/system/wazuh-dashboard.service
+ chown root:root /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/etc/rc.d/init.d/wazuh-dashboard
+ '[' 4.4.0 = 99.99.0 ']'
+ runuser wazuh-dashboard --shell=/bin/bash '--command=/build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/bin/opensearch-dashboards-plugin install https://packages-dev.wazuh.com/warehouse/test/4.4/ui/dashboard/wazuh-4.4.0-1.zip'
Attempting to transfer from https://packages-dev.wazuh.com/warehouse/test/4.4/ui/dashboard/wazuh-4.4.0-1.zip
Transferring 31864496 bytes....................
Transfer complete
Retrieving metadata from plugin archive
Extracting plugin archive
Extraction complete
Plugin installation complete
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/plugins/wazuh/ -exec chown wazuh-dashboard:wazuh-dashboard '{}' ';'
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/plugins/wazuh/ -type f -perm 644 -exec chmod 640 '{}' ';'
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/plugins/wazuh/ -type f -perm 755 -exec chmod 750 '{}' ';'
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/plugins/wazuh/ -type d -exec chmod 750 '{}' ';'
+ find /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64/usr/share/wazuh-dashboard/plugins/wazuh/ -type f -perm 744 -exec chmod 740 '{}' ';'
+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/redhat/brp-compress
+ /usr/lib/rpm/redhat/brp-strip /usr/bin/strip
+ /usr/lib/rpm/redhat/brp-strip-comment-note /usr/bin/strip /usr/bin/objdump
+ /usr/lib/rpm/redhat/brp-strip-static-archive /usr/bin/strip
+ /usr/lib/rpm/brp-python-bytecompile /usr/bin/python 1
+ /usr/lib/rpm/redhat/brp-python-hardlink
+ /usr/lib/rpm/redhat/brp-java-repack-jars
Processing files: wazuh-dashboard-4.4.0-wp.1880.x86_64
warning: File listed twice: /usr/share/wazuh-dashboard/node/bin/node
Provides: wazuh-dashboard = 4.4.0-wp.1880 wazuh-dashboard(x86-64) = 4.4.0-wp.1880
Requires(interp): /bin/sh /bin/sh /bin/sh /bin/sh /bin/sh
Requires(rpmlib): rpmlib(PartialHardlinkSets) <= 4.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1 rpmlib(CompressedFileNames) <= 3.0.4-1
Requires(pre): /usr/sbin/groupadd /usr/sbin/useradd /bin/sh
Requires(post): /bin/sh
Requires(preun): /sbin/service /bin/sh
Requires(postun): /sbin/service /bin/sh
Requires(posttrans): /bin/sh
Checking for unpackaged file(s): /usr/lib/rpm/check-files /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64
Wrote: /build/rpmbuild/SRPMS/wazuh-dashboard-4.4.0-wp.1880.src.rpm
Wrote: /build/rpmbuild/RPMS/x86_64/wazuh-dashboard-4.4.0-wp.1880.x86_64.rpm
Executing(%clean): /bin/sh -e /var/tmp/rpm-tmp.9Y9lbC
+ umask 022
+ cd /build/rpmbuild/BUILD
+ rm -fr /build/rpmbuild/BUILDROOT/wazuh-dashboard-4.4.0-wp.1880.x86_64
+ exit 0
Package wazuh-dashboard-4.4.0-wp.1880.x86_64.rpm.sha512 added to /stack/dashboard/rpm/output.
```

</details>

<details><summary>Wazuh dashboard Red Hat 7 install</summary>

```
[root@redhat7 vagrant]# ls -l /etc/init.d/
total 44
-rw-r--r--. 1 root root 18281 May 22  2020 functions
-rwxr-xr-x. 1 root root  4569 May 22  2020 netconsole
-rwxr-xr-x. 1 root root  7928 May 22  2020 network
-rw-r--r--. 1 root root  1160 May 12  2020 README
-rwxr-xr-x. 1 root root  2437 Oct 19  2017 rhnsd
[root@redhat7 vagrant]# yum localinstall wazuh-dashboard-4.4.0-wp.1880.x86_64.rpm -y
Loaded plugins: product-id, search-disabled-repos
Examining wazuh-dashboard-4.4.0-wp.1880.x86_64.rpm: wazuh-dashboard-4.4.0-wp.1880.x86_64
Marking wazuh-dashboard-4.4.0-wp.1880.x86_64.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package wazuh-dashboard.x86_64 0:4.4.0-wp.1880 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

==================================================
 Package         Arch   Version       Repository
                                             Size
==================================================
Installing:
 wazuh-dashboard x86_64 4.4.0-wp.1880 /wazuh-dashboard-4.4.0-wp.1880.x86_64
                                            709 M

Transaction Summary
==================================================
Install  1 Package

Total size: 709 M
Installed size: 709 M
Downloading packages:
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : wazuh-dashboard-4.4.0-wp.1880.x86_64                                                 1/1 
  Verifying  : wazuh-dashboard-4.4.0-wp.1880.x86_64                                                 1/1 
epel/x86_64/metalink                                                             |  28 kB  00:00:00     
epel/x86_64                                                                      | 4.7 kB  00:00:00     
epel/x86_64/group_gz                                                             |  97 kB  00:00:00     
epel/x86_64/updateinfo                                                           | 1.1 MB  00:00:00     
epel/x86_64/primary_db                                                           | 7.0 MB  00:00:00     

Installed:
  wazuh-dashboard.x86_64 0:4.4.0-wp.1880                                                                

Complete!
[root@redhat7 vagrant]# ls -l /etc/init.d/
total 48
-rw-r--r--. 1 root            root            18281 May 22  2020 functions
-rwxr-xr-x. 1 root            root             4569 May 22  2020 netconsole
-rwxr-xr-x. 1 root            root             7928 May 22  2020 network
-rw-r--r--. 1 root            root             1160 May 12  2020 README
-rwxr-xr-x. 1 root            root             2437 Oct 19  2017 rhnsd
-rwxr-x---. 1 wazuh-dashboard wazuh-dashboard  3682 Oct 21 12:29 wazuh-dashboard
[root@redhat7 vagrant]# 
```

</details>

<details><summary>Wazuh dashboard Red Hat 9 install</summary>

```
[root@redhat9 vagrant]# ls -l /etc/init.d/
total 0
[root@redhat9 vagrant]# yum localinstall wazuh-dashboard-4.4.0-wp.1880.x86_64.rpm -y
Last metadata expiration check: 20:07:14 ago on Thu 20 Oct 2022 04:27:07 PM UTC.
Dependencies resolved.
========================================================================================================
 Package                    Architecture      Version                     Repository               Size
========================================================================================================
Installing:
 wazuh-dashboard            x86_64            4.4.0-wp.1880               @commandline            172 M

Transaction Summary
========================================================================================================
Install  1 Package

Total size: 172 M
Installed size: 709 M
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                1/1 
  Running scriptlet: wazuh-dashboard-4.4.0-wp.1880.x86_64                                           1/1 
  Installing       : wazuh-dashboard-4.4.0-wp.1880.x86_64                                           1/1 
  Running scriptlet: wazuh-dashboard-4.4.0-wp.1880.x86_64                                           1/1 
  Verifying        : wazuh-dashboard-4.4.0-wp.1880.x86_64                                           1/1 
Installed products updated.

Installed:
  wazuh-dashboard-4.4.0-wp.1880.x86_64                                                                  

Complete!
[root@redhat9 vagrant]# ls -l /etc/init.d/
total 0
[root@redhat9 vagrant]#
```

</details>